### PR TITLE
Refactor engine process

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

Refactor engine process with follow mainly change:

* Add new class `EngineProcess` which is a wrapped of `Process`
* Move code from `ProcessBuilder` to `EngineProcess`

### _How was this patch tested?_
Pass CI

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @yaooqinn @pan3793

File: kyuubi-main/src/main/scala/org/apache/kyuubi/engine/EngineProcess.scala [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+import org.apache.commons.lang3.StringUtils.containsIgnoreCase
+
+import org.apache.kyuubi.KyuubiSQLException
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.engine.EngineProcess.{PROC_BUILD_LOGGER, UNCAUGHT_ERROR}
+import org.apache.kyuubi.util.NamedThreadFactory
+
+/**
+ * A wrapper of [[java.lang.Process]]
```
Comment (yaooqinn) [inline kyuubi-main/src/main/scala/org/apache/kyuubi/engine/EngineProcess.scala:33]: why it's a wrapper for [[java.lang.Process]] but we do not touch any of [[java.lang.Process]] in this trait?

File: kyuubi-main/src/main/scala/org/apache/kyuubi/engine/EngineProcess.scala [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+}
+object EngineProcess {
+  val UNCAUGHT_ERROR = new RuntimeException("Uncaught error")
+  val PROC_BUILD_LOGGER = new NamedThreadFactory("process-logger-capture", daemon = true)
+}
+
+/**
+ * Note that: This class is not thread safe.
+ */
+class EngineProcess(processBuilder: ProcessBuilder, conf: KyuubiConf, engineLog: File)
```
Comment (yaooqinn) [inline kyuubi-main/src/main/scala/org/apache/kyuubi/engine/EngineProcess.scala:80]: Will we have another concrete class for IEngineProcess

**Comment by yaooqinn** [comment]
TBH, I think this brings extra overhead for developers to create a new type of engine. Before they only need implement a new ProcBuilder by mimicking SparkProcBuilder with `java.lang.Process`, now they have to learn How EngineProcess works.

How about we only extract the log capturing part to a new trait? e.g. `EngineLogCapturer`？

**Comment by ulysses-you** [comment]
Seems a little overkill, close this.

**Pull request closed without merging**